src/main/java/net/dv8tion/jda/api/entities/Role.java

```diff
@@ -373,5 +373,47 @@ default String getIntegrationId()
         {
             return isIntegration() ? Long.toUnsignedString(getIntegrationIdLong()) : null;
         }
+
+        /**
+         * Whether this role can be acquired through a premium subscription purchase.
+         *
+         * @return True, if this is a subscription role
+         *
+         * @see    #getSubscriptionIdLong()
+         */
+        boolean hasSubscriptionListing();
+
+        /**
+         * The subscription listing id for this role.
+         *
+         * @return The listing id, or 0 if this role is not for a subscription listing
+         */
+        long getSubscriptionIdLong();
+
+        /**
+         * The subscription listing id for this role.
+         *
+         * @return The listing id, or null if this role is not for a subscription listing
+         */
+        @Nullable
+        default String getSubscriptionId()
+        {
+            return hasSubscriptionListing() ? Long.toUnsignedString(getSubscriptionIdLong()) : null;
+        }
+
+        /**
+         * Whether this role can be purchased by a user.
+         *
+         * @return True, if this role is purchasable
+         */
+        boolean isAvailableForPurchase();
```

Comment:
how is this different from `hasSubscriptionListing`?

Reply:
I have no idea. I don't even know how to setup roles like this yet. I just implemented these tags from the docs

![image](https://user-images.githubusercontent.com/18090140/218277982-1a901f34-dced-4977-b431-f052d93265a8.png)